app/frontend/src/features/auth/signup/AccountForm.tsx

```diff
@@ -94,231 +97,235 @@ export default function AccountForm() {
+      },
+      onError() {
+        window.scroll({ top: 0, behavior: "smooth" });
+      },
     }
-    setLoading(true);
+  );
 
-    try {
-      if (!authState.flowState) {
-        authActions.authError(
-          "Submitting account form without current signup flow in progress"
-        );
-      }
-      // authActions catches errors here
-      authActions.updateSignupState(
-        await service.auth.signupFlowAccount(
-          authState.flowState?.flowToken!,
-          lowercaseAndTrimField(data.username),
-          data.birthdate.format().split("T")[0],
-          data.gender,
-          acceptedTOS,
-          data.hostingStatus,
-          data.location.address,
-          data.location.lat,
-          data.location.lng,
-          data.location.radius
-          // TODO: password
-        )
-      );
-    } catch (err) {
-      authActions.authError(err.message);
+  const submit = handleSubmit(
+    (data: SignupAccountInputs) => {
+      mutation.mutate(data);
+    },
+    () => {
+      //location won't focus on error, so scroll to the top
+      if (errors.location) window.scroll({ top: 0, behavior: "smooth" });
     }
-    setLoading(false);
-  });
+  );
 
   return (
     <>
-      {loading ? (
-        <CircularProgress />
-      ) : (
-        <>
-          <form
-            className={`${authClasses.form} ${classes.firstForm}`}
-            onSubmit={submit}
-          >
-            <InputLabel className={authClasses.formLabel} htmlFor="username">
-              {USERNAME}
-            </InputLabel>
-            <TextField
-              className={authClasses.formField}
-              variant="standard"
-              id="username"
-              name="username"
-              fullWidth
-              inputRef={register({
-                pattern: {
-                  message: SIGN_UP_USERNAME_ERROR,
-                  value: usernameValidationPattern,
-                },
-                required: USERNAME_REQUIRED,
-                validate: async (username) => {
-                  const valid = await service.auth.validateUsername(username);
-                  return valid || USERNAME_TAKEN;
-                },
-              })}
-              helperText={errors?.username?.message ?? " "}
-              error={!!errors?.username?.message}
-            />
-            <InputLabel className={authClasses.formLabel} htmlFor="birthdate">
-              {SIGN_UP_BIRTHDAY}
-            </InputLabel>
-            <Datepicker
-              className={authClasses.formField}
-              control={control}
-              error={
-                //@ts-ignore Dayjs type breaks this
-                !!errors?.birthdate?.message
-              }
-              helperText={
-                //@ts-ignore
-                errors?.birthdate?.message ?? " "
+      {errors.location && (
+        //@ts-ignore - we register "location" but rhf thinks the error should be
+        //under location.address
```

Comment:
I wonder if updating to v7 of react-hook-form will fix this issue 🤔   it seems to infer the type wrong whenever the field has an object type apart from Date...